cluster-state-service/canary/pkg/wrappers/ecs.go

```diff
@@ -0,0 +1,161 @@
+// permissions and limitations under the License.
+
+package wrappers
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/ecs/ecsiface"
+	"github.com/pkg/errors"
+)
+
+var (
+	taskDefinitionSleep300 = "bloxCSSCanarySleep300"
+)
+
+// ECSWrapper defines methods to access wrapper methods to call ECS APIs
+type ECSWrapper interface {
+	CreateCluster(clusterName *string) error
+	DeleteCluster(clusterName *string) error
+	ListContainerInstances(clusterName *string) ([]*string, error)
+	DeregisterContainerInstances(clusterName *string, instanceARNs []*string) error
+	RegisterSleep360TaskDefinition() (string, error)
+}
+
+type ecsClientWrapper struct {
+	client ecsiface.ECSAPI
+}
+
+func NewECSWrapper(sess *session.Session) (ECSWrapper, error) {
+	if sess == nil {
+		return nil, errors.New("AWS session for initializing EC2 client cannot be nil")
+	}
+	ecsClient := ecs.New(sess)
+	return ecsClientWrapper{
+		client: ecsClient,
+	}, nil
+}
+
+func (wrapper ecsClientWrapper) CreateCluster(clusterName *string) error {
+	in := ecs.CreateClusterInput{
+		ClusterName: clusterName,
+	}
+
+	_, err := wrapper.client.CreateCluster(&in)
+	if err != nil {
+		return errors.Wrapf(err, "Error creating ECS cluster with name '%s'. ", clusterName)
+	}
+
+	return nil
+}
+
+func (wrapper ecsClientWrapper) DeleteCluster(clusterName *string) error {
+	in := ecs.DeleteClusterInput{
+		Cluster: clusterName,
+	}
+
+	_, err := wrapper.client.DeleteCluster(&in)
+	if err != nil {
+		return errors.Wrapf(err, "Error deleting ECS cluster with name '%s'. ", clusterName)
+	}
+
+	return nil
+}
+
+func (wrapper ecsClientWrapper) ListContainerInstances(clusterName *string) ([]*string, error) {
+	in := ecs.ListContainerInstancesInput{
+		Cluster: clusterName,
+	}
+
+	resp, err := wrapper.client.ListContainerInstances(&in)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error listing container instances for ECS cluster with name '%s'. ",
+			clusterName)
+	}
+
+	return resp.ContainerInstanceArns, nil
+}
+
+func (wrapper ecsClientWrapper) DeregisterContainerInstances(clusterName *string, instanceARNs []*string) error {
+	for _, instanceARN := range instanceARNs {
+		err := wrapper.deregisterContainerInstance(clusterName, instanceARN)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (wrapper ecsClientWrapper) deregisterContainerInstance(clusterName *string, instanceARN *string) error {
+	forceDeregister := true
+	in := ecs.DeregisterContainerInstanceInput{
+		Cluster:           clusterName,
+		ContainerInstance: instanceARN,
+		Force:             &forceDeregister,
+	}
+
+	_, err := wrapper.client.DeregisterContainerInstance(&in)
+	if err != nil {
+		return errors.Wrapf(err, "Error deregistering container instance with ARN '%s' in cluster with name '%s'",
+			instanceARN, clusterName)
+	}
+
+	return nil
+}
+
+func (wrapper ecsClientWrapper) RegisterSleep360TaskDefinition() (string, error) {
```

Comment:
My bad. Thanks, will fix it. I'll add a comment for making it configurable and get back to it. Have a whole bunch of things that has to be configurable.